derp/derp_client.go

```diff
@@ -175,7 +189,7 @@ func (c *Client) Recv(b []byte) (n int, err error) {
 			// require ack pongs.
 			continue
 		case frameRecvPacket:
-			return int(n), nil
+			return ReceivedPacket(b[:n]), nil
```

Comment:
This is a small allocation (a slice header, 24 bytes) but I'm not worried about it yet.

I'll care about performance later.

I'll find this back from profiling later if/when it matters.